api/users/serializers.py

```diff
@@ -381,6 +388,12 @@ def get_two_factor_enabled(self, obj):
         except TwoFactorUserSettings.DoesNotExist:
             return False
 
+    def get_secret(self, obj):
+        if self.context['request'].method in ('PUT', 'PATCH'):
+            two_factor_addon = obj.get_addon('twofactor')
+            if two_factor_addon and not two_factor_addon.is_confirmed:
```

Comment:
I thiiink technically this will also return it if you are only PATCHing one of the non-two factor settings and two factor happens to be on, which should be fine. But what I realized when I was thinking through this is that you should remove the self.context on L392 and only check for these items. Because if we reload the app and we're in the state where two factor is set but not confirmed, we will actually need the data. So I was dumb when I was trying to be extra secure.